### ISSUE TITLE
Allow template strings (with no interpolated expressions) to start off with literal types

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#16592]
I can't find the original issue (if there is one).

```ts
const c: "hello" = `hello`
```

Expected: No error.
Actual: Type 'string' is not assignable to type '"hello"'.

We should just allow people to use `NoSubstitutionTemplate`s the way that they're allowed to use regular strings.

**Comment by erkalicious** [comment]
This should also cover using `NoSubstitutionTemplate`s as literal types as in #18109

**Comment by erkalicious** [comment]
Is there any progress being made on this?
